tests/test_fitcube.py

```diff
@@ -529,5 +529,72 @@ def test_groupSettings(self):
+        fitter = Fitter(zinc_model_file, zinc_data_file)
+        self.assertEqual(1, len(fitter.getFitterSteps()))
+        fitter.setDiagnosticLevel(1)
+
+        # Rotation, scale, translation
+        transformationList = [[[ 0.0, 0.0, 0.0 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ math.pi * 20/180, 0.0, 0.0 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ math.pi * 135/180, 0.0, 0.0 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ math.pi * 250/180, math.pi * -45/180, 0.0 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ math.pi * 45/180, math.pi * 45/180, math.pi * 45/180 ], 1.0, [ 0.0, 0.0, 0.0 ]],
+                              [[ 0.0, 0.0, 0.0 ], 0.05, [ 0.0, 0.0, 0.0]],
+                              [[ math.pi * 70/180, math.pi * 10/180, math.pi * -300/180 ], 0.2, [ 0.0, 0.0, 0.0 ]],
+                              [[ 0.0, 0.0, 0.0 ], 1.0, [ 15.0, 15.0, 15.0 ]],
+                              [[ 0.0, 0.0, 0.0 ], 20.0, [ 50.0, 0.0, 10.0 ]],
+                              [[ math.pi * 90/180, math.pi * 200/180, math.pi * 5/180 ], 1.0, [ -10.0, -20.0, 100.0 ]],
+                              [[ math.pi * -45/180, math.pi * 120/180, math.pi * 10/180 ], 500.0, [ 100.0, 100.0, 100.0 ]]]
+
+        expectedAlignedNodes = [[-0.5690355951820659, 1.1070979208244695e-05, -0.40236892417087866],
+                                [-1.1077595833408616e-05, -0.5690355904946871, -0.4023689227447479],
+                                [1.1066291829453512e-05, 0.5690355885654408, -0.4023689255966489],
+                                [0.569035583878062, -1.1072908454692232e-05, -0.4023689241705181],
+                                [-0.5690355951822816, 1.1072995806778281e-05, 0.4023689241678401],
+                                [-1.107759604912495e-05, -0.5690355884780887, 0.40236892559397086],
+                                [1.10662916138482e-05, 0.5690355905820392, 0.4023689227420698],
+                                [0.5690355838778464, -1.1070891856158648e-05, 0.4023689241682007]]
+
+        for i in range(len(transformationList)):
+            fitter.load()
+            fieldmodule = fitter.getFieldmodule()
+            fieldcache = fieldmodule.createFieldcache()
+
+            modelCoordinates = fitter.getModelCoordinatesField()
+            rotation = transformationList[i][0]
+            scale = transformationList[i][1]
+            translation = transformationList[i][2]
+            modelCoordinatesTransformed = createFieldsTransformations(modelCoordinates, rotation, scale, translation)[0]
+            fieldassignment = modelCoordinates.createFieldassignment(modelCoordinatesTransformed)
+            fieldassignment.assign()
+
+            align = FitterStepAlign()
+            fitter.addFitterStep(align)
+            self.assertTrue(align.setAlignMarkers(True))
+            self.assertTrue(align.isAlignMarkers())
```

Comment:
This can also go out of the loop - only need one Align. Call align.run() each time will run it again from the new state.